app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -467,28 +464,18 @@ export default class TransactionStateManager extends EventEmitter {
     // Update state
     this._saveTxList(otherAccountTxs)
   }
+
   //
   //           PRIVATE METHODS
   //
 
-  // STATUS METHODS
-  // statuses:
-  //    - `'unapproved'` the user has not responded
-  //    - `'rejected'` the user has responded no!
-  //    - `'approved'` the user has approved the tx
-  //    - `'signed'` the tx is signed
-  //    - `'submitted'` the tx is sent to a server
-  //    - `'confirmed'` the tx has been included in a block.
-  //    - `'failed'` the tx failed for some reason, included on tx data.
-  //    - `'dropped'` the tx nonce was already used
-
   /**
-    @param {number} txId - the txMeta Id
-    @param {string} status - the status to set on the txMeta
-    @emits tx:status-update - passes txId and status
-    @emits ${txMeta.id}:finished - if it is a finished state. Passes the txMeta
-    @emits update:badge
-  */
+   * @param {number} txId - the txMeta Id
+   * @param {TransactionStatuses[keyof TransactionStatuses]} status - the status to set on the txMeta
```

Comment:
This doesn't seem right :thinking: The keys are upper-case strings, whereas the actual expected parameters are the lower-cased _values_. Unless I'm misunderstanding how this works?

Reply:
this is a property accessor, essentially you're getting all the values on the `TransactionStatuses` type that exist on any `keyof` the `TransactionStatuses` -- long story short it is a reference to the statuses. If i had done only `keyof TransactionStatuses` without the property accessor `TransactionStatuses` then it would be the upper case keys

Reply:
Ah of course! That makes sense, thanks. I did misread it.